maven doesn't generate javadoc.jar on archiving

Now maven doesn't generate javadoc.jar on archiving for each sub-project.
So we cannot see the javadoc on the internet, for example, javadoc.io (e.g. https://javadoc.io/doc/org.eclipse.leshan/leshan/1.3.0/index.html).

I made a quick fix for this problem:

```diff
diff --git a/pom.xml b/pom.xml
index d5f5d403..517377e9 100644
--- a/pom.xml
+++ b/pom.xml
@@ -542,6 +542,20 @@ Contributors:
                      </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <artifactId>maven-javadoc-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>attach-javadocs</id>
+                        <goals>
+                            <goal>jar</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <doclint>none</doclint>
+                </configuration>
+            </plugin>
             <plugin>
                 <artifactId>maven-enforcer-plugin</artifactId>
                 <executions>
```

But, as you can see, this disables doclint because now it raises doclint warnings and exits on generating javadoc.

What do you think about this?

Possibly this is off-topic, but `mvn javadoc:javadoc` reports some warnings and exits, so the command fails.

```
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for leshan 2.0.0-SNAPSHOT:
[INFO]
[INFO] leshan ............................................. SUCCESS [  1.251 s]
[INFO] leshan - core ...................................... FAILURE [  7.756 s]
[INFO] leshan - core californium .......................... SKIPPED
[INFO] leshan - server core ............................... SKIPPED
[INFO] leshan - server californium ........................ SKIPPED
[INFO] leshan - server redis .............................. SKIPPED
[INFO] leshan - client core ............................... SKIPPED
[INFO] leshan - client californium ........................ SKIPPED
[INFO] leshan - integration tests ......................... SKIPPED
[INFO] leshan - server demo ............................... SKIPPED
[INFO] leshan - client demo ............................... SKIPPED
[INFO] leshan - bootstrap server demo ..................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  9.868 s
[INFO] Finished at: 2020-12-17T23:56:05+09:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.1.1:javadoc (default-cli) on project leshan-core: An error has occurred in Javadoc report generation:
[ERROR] Exit code: 1 - /Users/moznion/ghq/github.com/eclipse/leshan/leshan-core/src/main/java/org/eclipse/leshan/core/tlv/TlvDecoder.java:160: warning: no @param for value
...
```

Probably it needs to fix javadoc?

I tried to run the same command on a container, to execute it in a clean environment, then it occurs the same issue.

The Dockerfile for that is the following:

```dockerfile
FROM openjdk:8u212-slim-buster

RUN apt -y update && \
        apt install -y curl && \
        curl -L -o /opt/apache-maven-3.6.0-bin.tar.gz https://archive.apache.org/dist/maven/maven-3/3.6.0/binaries/apache-maven-3.6.0-bin.tar.gz && \
        tar --directory=/opt -zxf /opt/apache-maven-3.6.0-bin.tar.gz
ENV PATH $PATH:/opt/apache-maven-3.6.0/bin
```

and execute the following command on the repository root on the host machine:

```shell
$ docker build . -t leshan-java:latest
$ docker run --rm -v $(pwd):/leshan -w /leshan leshan-java:latest sh -c 'java -version && mvn -version && mvn install javadoc:javadoc'
```

This outputs like the following result:

```
openjdk version "1.8.0_212"
OpenJDK Runtime Environment (build 1.8.0_212-b04)
OpenJDK 64-Bit Server VM (build 25.212-b04, mixed mode)
Apache Maven 3.6.0 (97c98ec64a1fdfee7767ce5ffb20918da4f719f3; 2018-10-24T18:41:47Z)
Maven home: /opt/apache-maven-3.6.0
Java version: 1.8.0_212, vendor: Oracle Corporation, runtime: /usr/local/openjdk-8/jre
Default locale: en, platform encoding: UTF-8
OS name: "linux", version: "4.19.121-linuxkit", arch: "amd64", family: "unix"
[INFO] Scanning for projects...
...snip...
[INFO] --- maven-javadoc-plugin:3.1.1:javadoc (default-cli) @ leshan-core ---
[INFO]
1 error
100 warnings
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for leshan 2.0.0-SNAPSHOT:
[INFO]
[INFO] leshan ............................................. SUCCESS [01:28 min]
[INFO] leshan - core ...................................... FAILURE [01:21 min]
[INFO] leshan - core californium .......................... SKIPPED
[INFO] leshan - server core ............................... SKIPPED
[INFO] leshan - server californium ........................ SKIPPED
[INFO] leshan - server redis .............................. SKIPPED
[INFO] leshan - client core ............................... SKIPPED
[INFO] leshan - client californium ........................ SKIPPED
[INFO] leshan - integration tests ......................... SKIPPED
[INFO] leshan - server demo ............................... SKIPPED
[INFO] leshan - client demo ............................... SKIPPED
[INFO] leshan - bootstrap server demo ..................... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  04:23 min
[INFO] Finished at: 2020-12-21T15:29:25Z
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-javadoc-plugin:3.1.1:javadoc (default-cli) on project leshan-core: An error has occurred in Javadoc report generation:
[ERROR] Exit code: 1 - /leshan/leshan-core/src/main/java/org/eclipse/leshan/core/Link.java:84: warning: no @param for <T>
[ERROR]     public <T> Link(String url, Map<String, T> attributes, Class<T> clazz) {
[ERROR]                ^
[ERROR] /leshan/leshan-core/src/main/java/org/eclipse/leshan/core/Link.java:84: warning: no @param for clazz
[ERROR]     public <T> Link(String url, Map<String, T> attributes, Class<T> clazz) {
[ERROR]                ^
[ERROR] /leshan/leshan-core/src/main/java/org/eclipse/leshan/core/attributes/AttributeSet.java:176: warning: no @return
[ERROR]     public static AttributeSet parse(String uriQueries) {
...snip...
```

Does it reproduce this in your environment?

--- a/leshan-bsserver-demo/pom.xml
+++ b/leshan-bsserver-demo/pom.xml
@@ -122,6 +122,12 @@ Contributors:
                     <skip>true</skip>
                 </configuration>
             </plugin>
+            <plugin>
+                <artifactId>maven-javadoc-plugin</artifactId>
+                <configuration>
+                    <failOnWarnings>false</failOnWarnings>
+                </configuration>
+            </plugin>
         </plugins>
     </build>
 </project>

--- a/leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/X509Util.java
+++ b/leshan-client-cf/src/main/java/org/eclipse/leshan/client/californium/X509Util.java
@@ -80,7 +80,6 @@ public class X509Util {
      * @param certPath a certificate path to eventually truncate
      * @param trustedCertificates list of trusted certificates
      * @return a certPath without trusted any trusted certificates
-     * @throws CertificateEncodingException
      */
     public static CertPath truncateToFirstTrustedCert(CertPath certPath, X509Certificate[] trustedCertificates)
             throws CertificateEncodingException {
@@ -100,9 +99,8 @@ public class X509Util {
      * @param certPath a certificate path to extend
      * @param certificate to add at the end
      * @return a certPath ending with given certificate
-     * @throws CertificateEncodingException
      */
-    public static CertPath add(CertPath certPath, X509Certificate certificate) throws CertificateEncodingException {
+    public static CertPath add(CertPath certPath, X509Certificate certificate) {
         List<X509Certificate> certificates = CertPathUtil.toX509CertificatesList(certPath.getCertificates());
         certificates.add(certificate);
         return CertPathUtil.generateCertPath(certificates);
@@ -114,10 +112,8 @@ public class X509Util {
      * @param certificate certificate to search
      * @param certificates to search
      * @return {@code true}, if certificate is contained, {@code false}, otherwise.
-     * @throws CertificateEncodingException if encoding a certificate failed!
      */
-    public static boolean contains(X509Certificate certificate, X509Certificate[] certificates)
-            throws CertificateEncodingException {
+    public static boolean contains(X509Certificate certificate, X509Certificate[] certificates) {
         for (X509Certificate trust : certificates) {
             if (certificate.equals(trust)) {
                 return true;

--- a/leshan-core/src/main/java/org/eclipse/leshan/core/model/ObjectLoader.java
+++ b/leshan-core/src/main/java/org/eclipse/leshan/core/model/ObjectLoader.java
@@ -107,7 +107,6 @@ public class ObjectLoader {
      * @param streamName A name for the stream used for logging only
      * 
      * @throws InvalidDDFFileException if DDF file is invalid
-     * @throws IOException
      */
     public static List<ObjectModel> loadDdfFile(InputStream input, String streamName)
             throws InvalidDDFFileException, IOException {
@@ -125,7 +124,6 @@ public class ObjectLoader {
      * 
      * @throws InvalidDDFFileException if DDF file is invalid
      * @throws InvalidModelException if model is invalid
-     * @throws IOException
      * 
      * @since 1.1
      */
@@ -146,7 +144,6 @@ public class ObjectLoader {
      * 
      * @throws InvalidDDFFileException if DDF file is invalid
      * @throws InvalidModelException if model is invalid
-     * @throws IOException
      */
     public static List<ObjectModel> loadDdfFile(InputStream input, String streamName, DDFFileParser ddfFileParser,
             ObjectModelValidator modelValidator) throws InvalidModelException, InvalidDDFFileException, IOException {
@@ -171,7 +168,6 @@ public class ObjectLoader {
      * 
      * @throws InvalidDDFFileException if DDF file is invalid
      * @throws InvalidModelException if model is invalid
-     * @throws IOException
      */
     public static List<ObjectModel> loadDdfResources(String path, String[] filenames)
             throws IOException, InvalidModelException, InvalidDDFFileException {
@@ -190,7 +186,6 @@ public class ObjectLoader {
      * 
      * @throws InvalidDDFFileException if DDF file is invalid
      * @throws InvalidModelException if model is invalid
-     * @throws IOException
      * 
      * @since 1.1
      */
@@ -215,7 +210,6 @@ public class ObjectLoader {
      * 
      * @throws InvalidDDFFileException if DDF file is invalid
      * @throws InvalidModelException if model is invalid
-     * @throws IOException
      */
     public static List<ObjectModel> loadDdfResources(String path, String[] filenames, DDFFileParser ddfFileParser,
             ObjectModelValidator modelValidator) throws IOException, InvalidModelException, InvalidDDFFileException {
@@ -245,8 +239,7 @@ public class ObjectLoader {
      * @param paths An array of paths to DDF files.
      * 
      * @throws InvalidDDFFileException if DDF file is invalid
-     * @throws IOException
-     * @throws InvalidModelException
+     * @throws InvalidModelException if model is invalid
      */
     public static List<ObjectModel> loadDdfResources(String[] paths)
             throws InvalidDDFFileException, IOException, InvalidModelException {
@@ -264,7 +257,6 @@ public class ObjectLoader {
      *        model if you already trust it.
      * 
      * @throws InvalidDDFFileException if DDF file is invalid
-     * @throws IOException
      */
     public static List<ObjectModel> loadDdfResources(String[] paths, boolean validate)
             throws IOException, InvalidModelException, InvalidDDFFileException {
@@ -284,7 +276,6 @@ public class ObjectLoader {
      *        {@link DefaultObjectModelValidator}. If {@code null} then there will be no validation.
      * 
      * @throws InvalidDDFFileException if DDF file is invalid
-     * @throws IOException
      */
     public static List<ObjectModel> loadDdfResources(String[] paths, DDFFileParser ddfFileParser,
             ObjectModelValidator modelValidator) throws IOException, InvalidModelException, InvalidDDFFileException {

--- a/leshan-core/src/main/java/org/eclipse/leshan/core/model/URN.java
+++ b/leshan-core/src/main/java/org/eclipse/leshan/core/model/URN.java
@@ -42,7 +42,6 @@ public class URN {
     /**
      * Return URN "kind" from object id.
      * 
-     * @param objectId
      * @return {@link #OMA_LABEL}, {@link #EXT_LABEL} or {@link #X_LABEL} for valid object id. {@link #INVALID_LABEL} is
      *         returned for invalid id and {@link #RESERVED_LABEL} for reserved range.
      * 

--- a/leshan-core/src/main/java/org/eclipse/leshan/core/request/DiscoverRequest.java
+++ b/leshan-core/src/main/java/org/eclipse/leshan/core/request/DiscoverRequest.java
@@ -59,7 +59,6 @@ public class DiscoverRequest extends AbstractDownlinkRequest<DiscoverResponse> {
      * path.
      *
      * @param path the path of the LWM2M node to discover
-     * @param coapRequest the underlying request
      * @exception InvalidRequestException if the path is not valid.
      */
     public DiscoverRequest(String path) throws InvalidRequestException {

--- a/leshan-server-demo/pom.xml
+++ b/leshan-server-demo/pom.xml
@@ -137,6 +137,12 @@ Contributors:
                 </configuration>
             </plugin>
             <plugin>
+                    <artifactId>maven-javadoc-plugin</artifactId>
+                    <configuration>
+                        <failOnWarnings>false</failOnWarnings>
+                    </configuration>
+                </plugin>
+            <plugin>
                 <groupId>org.revapi</groupId>
                 <artifactId>revapi-maven-plugin</artifactId>
                 <configuration>

--- a/pom.xml
+++ b/pom.xml
@@ -384,6 +384,7 @@ Contributors:
                         <excludePackageNames>org.eclipse.leshan.core.util</excludePackageNames>
                         <quiet>true</quiet>
                         <doclint>${javadoc.doclint}</doclint>
+                        <failOnWarnings>true</failOnWarnings>
                     </configuration>
                 </plugin>
                 <plugin>

--- a/pom.xml
+++ b/pom.xml
@@ -128,14 +128,14 @@ Contributors:
                 <jdk>[1.8,)</jdk>
             </activation>
             <properties>
-                <jdk8orGreater>true</jdk8orGreater>
+                <javadoc.doclint>all,-missing</javadoc.doclint>
             </properties>
         </profile>
         <profile>
             <!-- This profile activate the usage of toolchain to compile using JDK 1.7-->
             <id>useToolchain</id>
             <properties>
-                <jdk8orGreater>false</jdk8orGreater>
+                <javadoc.doclint></javadoc.doclint>
             </properties>
             <build>
                 <plugins>
@@ -154,23 +154,6 @@ Contributors:
                                     <version>1.7</version>
                                 </jdk>
                             </toolchains>
-                        </configuration>
-                    </plugin>
-                </plugins>
-            </build>
-        </profile>
-        <profile>
-            <id>configure-doclint</id>
-            <activation>
-                <property><name>jdk8orGreater</name><value>true</value></property>
-            </activation>
-            <build>
-                <plugins>
-                    <plugin>
-                        <groupId>org.apache.maven.plugins</groupId>
-                        <artifactId>maven-javadoc-plugin</artifactId>
-                        <configuration>
-                            <doclint>all,-missing</doclint>
                         </configuration>
                     </plugin>
                 </plugins>
@@ -400,6 +383,7 @@ Contributors:
                     <configuration>
                         <excludePackageNames>org.eclipse.leshan.core.util</excludePackageNames>
                         <quiet>true</quiet>
+                        <doclint>${javadoc.doclint}</doclint>
                     </configuration>
                 </plugin>
                 <plugin>

